tests/test_hilbert_sort.py

```diff
@@ -0,0 +1,43 @@
+"""Intended to test the custom_ops.HilbertSort. Note that this does not compare against the
+the tests/common.py::hilbert_sort function which produces a different permutation. This simply verifies that
+sorting will produce more compact blocks.
+"""
+import numpy as np
+import pytest
+from numpy.typing import NDArray
+
+from timemachine.lib import custom_ops
+from timemachine.potentials.jax_utils import delta_r
+from timemachine.testsystems.dhfr import setup_dhfr
+
+pytestmark = [pytest.mark.memcheck]
+
+
+def get_max_block_distances(coords: NDArray, box: NDArray, block_size: int) -> NDArray:
+    """Compute the max distance between all particles within each block"""
+    # Act on a copy to avoid modifying original coordinates
+    coords = coords.copy()
+    N = coords.shape[0]
+    num_blocks = (N + block_size - 1) // block_size
+    block_distances = []
+    for bidx in range(num_blocks):
+        start_idx = bidx * block_size
+        end_idx = min((bidx + 1) * block_size, N)
+        block = coords[start_idx:end_idx]
+        block_distances.append(np.max(delta_r(block[:, None], block[None, :], box=box)))
+    return np.array(block_distances)
+
+
+@pytest.mark.parametrize("block_size", [8, 16, 32])
+def test_hilbert_sort_dhfr(block_size):
+    _, _, coords, box = setup_dhfr()
+    distances = get_max_block_distances(coords, box, block_size)
+    unsorted_mean_dist = np.mean(distances)
+
+    sorter = custom_ops.HilbertSort(coords.shape[0])
+    perm = sorter.sort(coords, box)
+    sorted_coords = coords[perm]
+    sorted_dists = get_max_block_distances(sorted_coords, box, block_size)
+    sorted_mean_dist = np.mean(sorted_dists)
+    # The mean maximum distance within hilbert curve sorted blocks should be lower than the unsorted coords
+    assert sorted_mean_dist < unsorted_mean_dist
```

Comment:
Could this be strengthened? At present it seems like this would pass 1/8 of the time even if the sorted and unsorted mean distances have the same distribution.

Reply:
Good question, I couldn't think of something obvious when I did this. Will take a look through our hilbert curve python library to see how they verify their code.

Reply:
Looked through the tests here https://github.com/galtay/hilbertcurve/blob/main/tests/test_hilbertcurve.py nothing stood out. Actually makes me trust the library less, though we don't use it in practice. 

Any suggestions?

Reply:
I guess I was just thinking to empirically choose a factor by which the hilbert-sorted distances should be smaller for this test. Testing that the code is generating a correct hilbert curve seems more difficult; don't immediately have good ideas for the latter.

Reply:
Strengthened to be less than half the distance of the unsorted: https://github.com/proteneer/timemachine/pull/1126/commits/938610b38b13764b7875ebe38b74efb629be079b